sdks/java/core/src/main/java/org/apache/beam/sdk/io/WriteFiles.java

```diff
@@ -394,6 +418,7 @@ public void startBundle(StartBundleContext c) {
 
     @ProcessElement
     public void processElement(ProcessContext c, BoundedWindow window) throws Exception {
+      sink.getDynamicDestinations().setSideInputAccessorFromProcessContext(c);
```

Comment:
<!--thread_id:cc_128889794_t; commit:3708e20aa6a660b2f1e3519656675abc8887ab17; resolved:1-->
<!--section:context-quote-->
> **reuvenlax** wrote:
> Alternatively we could make SideInputAccessor a parameter to every method on DynamicDestinations. I believe we considered this when we did BigQueryIO and rejected it. @jkff might remember why we rejected it. (possibly because it seemed like it made the API awkward to always pass a parameter the user might not use).

<!--section:body-->
Making it a parameter to every method is equivalent to making it a parameter to a single method that returns the object with it bound. Both of my examples are that. But adding a factory layer for users can be annoying in Java if it isn't integrated carefully. I won't block on this; I was really just on the PR to check the translation stuff.